File: backend/app/services/guitai_song_service.py
from app.schemas.guitai_song import (
    GuitAISongResponse,
    SpotifyTrackInput,
)


def prepare_spotify_song(
    track: SpotifyTrackInput,
) -> GuitAISongResponse:
    return GuitAISongResponse(
        id=f"spotify-{track.id}",
        source="spotify",
        source_id=track.id,
        title=track.title,
        artist=track.artist,
        album=track.album,
        image_url=track.image_url,
        source_url=track.spotify_url,
        duration_ms=track.duration_ms,
        explicit=track.explicit,
        status="waiting_for_lesson",
        lesson_available=False,
    )

File: src/services/guitaiSongService.ts
import type {
  SpotifyTrack,
} from './spotifyService'

import type {
  GuitAISong,
} from '../types/guitaiSong'


const API_BASE_URL =
  import.meta.env.VITE_API_URL ??
  'http://localhost:8000'


async function prepareSpotifyTrack(
  track: SpotifyTrack
): Promise<GuitAISong> {
  const response =
    await fetch(
      `${API_BASE_URL}/songs/prepare`,
      {
        method: 'POST',

        headers: {
          'Content-Type':
            'application/json',
        },

        body: JSON.stringify({
          id: track.id,
          title: track.title,
          artist: track.artist,
          album: track.album,
          image_url:
            track.image_url,
          spotify_url:
            track.spotify_url,
          duration_ms:
            track.duration_ms,
          explicit:
            track.explicit,
        }),
      }
    )

  if (!response.ok) {
    throw new Error(
      'Não foi possível preparar esta música.'
    )
  }

  const song =
    await response.json() as GuitAISong

  return song
}


export const guitaiSongService = {
  prepareSpotifyTrack,
}